### PR TITLE
Add LESS support

### PR DESCRIPTION
**Pull request by ChrisOMetz**
Added support for less as requested in
https://github.com/RehanSaeed/ASP.NET-MVC-Boilerplate/issues/51

### CODE REVIEW

**Comment by ChrisOMetz** [comment]
Hi Rehan,
I was not able to test the wizard itself as I am completely blank about how to do this. Furthermore this is my first pull request ever. So beg your pardon when I did something wrong.

So long
Christian

**Comment by RehanSaeed** [comment]
Good effort, you did a pretty good job. There is instructions on how to build and run the solution [here](https://github.com/RehanSaeed/ASP.NET-MVC-Boilerplate/blob/master/HOW%20TO%20BUILD.md). I made some comments for each file.

File: Source/MVC6/Boilerplate.Web.Mvc6.Sample/config.json [hunk 1]
```diff
@@ -2,12 +2,14 @@
   "AppSettings": {
     "SiteShortTitle": "Boilerplate",
     "SiteTitle": "ASP.NET MVC Boilerplate"
-  },
+  }
   // $Start-ApplicationInsights$
+  ,
```
Comment (RehanSaeed) [inline Source/MVC6/Boilerplate.Web.Mvc6.Sample/config.json]: Commas moved onto their own lines.

File: Source/MVC6/Boilerplate.Web.Mvc6.Sample/gulpfile.js [hunk 2]
```diff
@@ -35,8 +35,13 @@ var gulp = require("gulp"),
     merge = require("merge-stream"),            // Merges one or more gulp streams into one (https://www.npmjs.com/package/merge-stream/)
     psi = require("psi"),                       // Google PageSpeed performance tester (https://www.npmjs.com/package/psi/)
     rimraf = require("rimraf"),                 // Deletes files and folders (https://www.npmjs.com/package/rimraf/)
+    // $Start-Less$
+    less = require('gulp-less'),                // Compile LESS to CSS (https://www.npmjs.com/package/gulp-less/)
+    // $Start-Less$
```
Comment (RehanSaeed) [inline Source/MVC6/Boilerplate.Web.Mvc6.Sample/gulpfile.js]: The ending needs to be 
```
// $End-Less$
```

File: Source/MVC6/Boilerplate.Web.Mvc6.Sample/gulpfile.js [hunk 3]
```diff
@@ -35,8 +35,13 @@ var gulp = require("gulp"),
     merge = require("merge-stream"),            // Merges one or more gulp streams into one (https://www.npmjs.com/package/merge-stream/)
     psi = require("psi"),                       // Google PageSpeed performance tester (https://www.npmjs.com/package/psi/)
     rimraf = require("rimraf"),                 // Deletes files and folders (https://www.npmjs.com/package/rimraf/)
+    // $Start-Less$
+    less = require('gulp-less'),                // Compile LESS to CSS (https://www.npmjs.com/package/gulp-less/)
+    // $Start-Less$
+    // $End-Sass$
```
Comment (RehanSaeed) [inline Source/MVC6/Boilerplate.Web.Mvc6.Sample/gulpfile.js]: Start needs to be:
```
// $Start-Sass$
```

File: Source/MVC6/Boilerplate.Web.Mvc6.Sample/gulpfile.js [hunk 4]
```diff
@@ -254,8 +296,8 @@ gulp.task("clean-fonts", function (cb) {
 gulp.task("clean-js", function (cb) {
     return rimraf(paths.js, cb);
 });
-// $Start-CshtmlMinification$
```
Comment (RehanSaeed) [inline Source/MVC6/Boilerplate.Web.Mvc6.Sample/gulpfile.js]: Blank space matters here. If someone takes out the CshtmlMinification feature.

File: Source/MVC6/Boilerplate.Web.Mvc6.Sample/gulpfile.js [hunk 5]
```diff
@@ -269,18 +311,27 @@ gulp.task("clean-html", function (cb) {
  */
 gulp.task("clean", ["clean-css", "clean-fonts", "clean-js"]);
 
+// $Start-Less$
+/*
+ * Report warnings and errors in your CSS files (lint them) under the Styles folder.
+ */
+// End-Less$
```
Comment (RehanSaeed) [inline Source/MVC6/Boilerplate.Web.Mvc6.Sample/gulpfile.js]: Missing '$'.

File: Source/MVC6/Boilerplate.Web.Mvc6.Sample/gulpfile.js [hunk 6]
```diff
@@ -459,6 +515,19 @@ gulp.task("optimize-images", function () {
         .pipe(sizeAfter());                 // Write the size of the file to the console after minification.
 });
 
+// $Start-Less$
```
Comment (RehanSaeed) [inline Source/MVC6/Boilerplate.Web.Mvc6.Sample/gulpfile.js]: There is already a watch task.

File: Source/MVC6/Boilerplate.Web.Mvc6.TagHelpers/project.json [hunk 7]
```diff
@@ -1,5 +1,5 @@
 ﻿{
```
Comment (RehanSaeed) [inline Source/MVC6/Boilerplate.Web.Mvc6.TagHelpers/project.json]: This file does not need to change.

File: Source/MVC6/Boilerplate.Web.Mvc6.Sample/project.json [hunk 8]
```diff
@@ -3,10 +3,10 @@
   "version": "1.0.0-*",
```
Comment (RehanSaeed) [inline Source/MVC6/Boilerplate.Web.Mvc6.Sample/project.json]: This file does not need to change.

File: Source/MVC6/Boilerplate.Web.Mvc6/project.json [hunk 9]
```diff
@@ -1,5 +1,5 @@
 {
```
Comment (RehanSaeed) [inline Source/MVC6/Boilerplate.Web.Mvc6/project.json]: This file does not need to change.

File: Source/Solution.sln [hunk 10]
```diff
@@ -72,7 +72,6 @@ Global
 		{E0B062AF-E0B4-4F72-A284-22468F9C2A4A}.Release|Any CPU.ActiveCfg = Release|Any CPU
```
Comment (RehanSaeed) [inline Source/Solution.sln]: This file does not need to change.

File: Privacy/ReferrerMetaTagFeature.cs [hunk 11]
```diff
@@ -0,0 +1,62 @@
+﻿namespace Boilerplate.Wizard.Features
```
Comment (RehanSaeed) [inline Privacy/ReferrerMetaTagFeature.cs]: This file does not need to change.

File: Source/MVC6/Boilerplate.Web.Mvc6.Sample/bower.json [hunk 12]
```diff
@@ -2,7 +2,12 @@
   "name": "asp.net-mvc-boilerplate",
   "private": true,
   "dependencies": {
+    // $Start-Sass$
```
Comment (RehanSaeed) [inline Source/MVC6/Boilerplate.Web.Mvc6.Sample/bower.json]: The start and end comments need to be changed to this in all cases:

```
// $Start-StylesheetLanguage-Less$
// $End-StylesheetLanguage-Less$


// $Start-StylesheetLanguage-Sass$
// $End-StylesheetLanguage-Sass$
```

File: Source/MVC6/Boilerplate.Web.Mvc6.Sample/bower.json [hunk 13]
```diff
@@ -2,7 +2,12 @@
   "name": "asp.net-mvc-boilerplate",
   "private": true,
   "dependencies": {
+    // $Start-Sass$
     "bootstrap-sass": "3.3.5",
+    // $End-Sass$
+    // $Start-Less$
+    "bootstrap-less": "3.3.4",
```
Comment (RehanSaeed) [inline Source/MVC6/Boilerplate.Web.Mvc6.Sample/bower.json]: Looks like the LESS version only goes up to version 3.3.4.

**Comment by ChrisOMetz** [comment]
Fixed the markup problems and tested the wizard. Thank you for your support in there. The How-to page was helpfull.
Next I am struggling with the problem of the changed files (without real changes). Is there a way to revert just them?

**Comment by RehanSaeed** [comment]
I'm not entirely sure. I suspect that if you create a new change so they are reverted to how they were at the beginning the changes will not show here.

**Pull request closed without merging**